README.md

```diff
@@ -145,6 +145,26 @@ func case5(x int) {                                //-|
 | —format | Format of the diff coverage report, one of: html, json, markdown |
 | —excludes | Exclude files for diff coverage inspection |
 
+## Miscellaneous
+
+### How to calculate diff coverage
+
+There are mainly there steps to calculate diff coverage for a module.
+
+1. Use `go test` to generate cover profiles for the module because `gocover` highly relies on cover profiles generated by it.
+2. Generate git diff changes compared current branch with master/main branch.
+3. Loop over each line from the diff changes, and reverse lookup the profile block from the [cover profile](https://pkg.go.dev/golang.org/x/tools@v0.1.10/cover) in the step 1. The `Count` field of cover profile indicates whether this code line is covered by unit test or not.
+
+### About `_test.go`
+
+`gocover` highly relies on cover profiles generated by `go test`. `go test` generates cover profiles for a specific package at circumstance that there is at least one `_test.go` in the package.
+
+So there would be two situations:
+1. A new package has `_test.go` file. `go test` can generate cover profile demandly, then `gocover` will generate diff coverage for the package.
+2. A new package has no `_test.go` file, `go test` won't generate any cover profile for the package, `gocover` won't generate any diff coverage for the package.
+
+If a package needs diff coverage result, please make sure that a least one `_test.go` exists, even though its contents are emtpy.
```

Comment:
suppose I have a package A, which is referenced by package B. In package B, I have `_test.go` while no for A. So, what's the result of the following scenarios:

1. no `_test.go` in package A
2. have an empty `_test.go` in package B

Also, could you clarify what does "empty `_test.go`" mean here?

For example:

```
package a
```

or

```
package a

import "testing"

func TestFoo(t *testing.T) {

}
```

Reply:
Further more, if the package A is purely generated from code (like mockgen), why should we care about the coverage for this package? Therefore, I think we would better to frame the usage and description as:

1. `gocover` relies on `go cover` to generate test coverage
2. `go cover` relies on `_test.go` under a package to calculate coverage for the package (not crossing the package)
3. if we don't care about the test coverage for a package, it's fine to have no `_test.go` under the package
4. if we want to count the package into test coverage, we should at least create one `_test.go` under the package

For item 3, please also clarify what's the effect to the repo test coverage if the package A has no tests

Reply:
It only tests the package B. no test will run against package A.

Reply:
That's incorrect behavior for the generated code. And is fixed now.